app/views/steps/_new.html.erb

```diff
@@ -1,5 +1,5 @@
 <div class="well">
-  <%= bootstrap_form_for(@step, url: protocol_steps_path(@protocol, format: :json), remote: true, authenticity_token: true, multipart: true, data: { role: "new-step-form", type: :json }) do |f| %>
+  <%= bootstrap_form_for(@step, url: protocol_steps_path(@protocol, format: :json), authenticity_token: true, multipart: true, data: { role: "new-step-form", type: :json }) do |f| %>
```

Comment:
You can actually leave remote: true, that way you don't have to manually make AJAX request in JS, but instead, just handle success/error responses. Final result -> less code.